script/run-isolated-tests.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+# Get the list of subdirectories that contain apps and
+# extract substrings from the paths for the following regex.
+APP_SUBPATHS="$(find src -name manifest.json | sed -E 's/\/manifest.json//g' | xargs | sed 's/ /|/g')"
+
+# Match changed files to the app paths determined above
+FILES_CHANGED="$(git diff --name-only master)"
+APP_SUBPATHS_CHANGED="$(echo $FILES_CHANGED | grep -oE "($APP_SUBPATHS)")"
+NUM_APPS_CHANGED=$(echo "$APP_SUBPATHS_CHANGED" | wc -l)
+
+# Run tests only within the apps that have changed.
```

Comment:
If I understand this correctly, the tests will run on apps that have changed in master even if the PR didn't modify those apps.  It may be worth making a note of that.